roles/pulp/tasks/main.yml

```diff
@@ -16,17 +16,18 @@
   # pipefail is not needed due to grep's rc, but it resolves ansible-lint 306
   shell: |
     set -o pipefail
-    systemctl show-environment | grep -E '^PATH='
+    systemctl show-environment | grep -E '^PATH=' | sed 's/PATH=//g'
   args:
     # Debian defaults to bourne, but it doesn't understand pipefail.
     executable: /bin/bash
   changed_when: false
   register: systemd_show_env_path
-  check_mode: yes
+  when: not ansible_check_mode
```

Comment:
So we need this even in check_mode to determine the file contents properly.

I think the solution is:
`check_mode: no`

per:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_checkmode.html

Also, we have a larger task to support check_mode:
https://pulp.plan.io/issues/6230

If we fix 6254 but not 6230, his check_mode would probably have no errors, but would still report multiple changes done.

Reply:
agreed, I'll put this PR in WIP, and start to look at 6230